### PR TITLE
PeerDAS: Batch columns verifications

### PR DESCRIPTION
**Pull request by nalepae**
This PR transforms `NewColumnVerifier` (1 column) into `NewDataColumnsVerifier` (multiple columns, `Data` has been added for consistency).

The purpose of this PR is to batch data columns verification, and especially `VerifyDataColumnsSidecarKZGProofs`, which is much more efficient when called by batch than when called columns by columns.

On my computer (ARM M-2), duration of KZG verification assuming a constant blob count per block, for 32 blocks:
- column by column: `~12-13 s`
- batch (all columns in the batch of 32 blocks): `~700 ms`

==> `x18` (!) improvement.

- [x] Able to sync a super node from genesis on `peerdas-devnet-3`
- [x] Able to sync a full node from genesis on `peerdas-devnet-3` (in progress)

### CODE REVIEW

File: beacon-chain/verification/data_column.go [hunk 1]
```diff
@@ -69,26 +68,32 @@ var (
 	ErrColumnIndexInvalid = errors.New("incorrect column sidecar index")
 )
 
-type RODataColumnVerifier struct {
+type RODataColumnsVerifier struct {
```
Comment (nisdas) [inline beacon-chain/verification/data_column.go:71]: Why is this being changed ? We already have a batch verifier for the cases where we need to modify multiple columns at once

File: beacon-chain/sync/data_columns_sampling.go [hunk 2]
```diff
@@ -538,9 +538,11 @@ func verifyColumn(
 		return false
 	}
 
-	vf := columnVerifier(roDataColumn, verification.SamplingColumnSidecarRequirements)
+	// TODO: Once peer sampling is used, we should verify all sampled data columns in a single batch.
+	verifier := dataColumnsVerifier([]blocks.RODataColumn{roDataColumn}, verification.SamplingColumnSidecarRequirements)
```
Comment (nisdas) [inline beacon-chain/sync/data_columns_sampling.go:542]: This is a nit, but in go its always more preferable for shorter variable names (vf vs verifier). This was done in a few places in this PR

File: beacon-chain/sync/validate_data_column.go [hunk 3]
```diff
@@ -141,17 +143,22 @@ func (s *Service) validateDataColumn(ctx context.Context, pid peer.ID, msg *pubs
...
 	}
 
-	verifiedRODataColumn, err := verifier.VerifiedRODataColumn()
+	verifiedRODataColumns, err := verifier.VerifiedRODataColumns()
 	if err != nil {
 		return pubsub.ValidationReject, err
 	}
 
-	msg.ValidatorData = verifiedRODataColumn
+	if len(verifiedRODataColumns) != 1 {
```
Comment (nisdas) [inline beacon-chain/sync/validate_data_column.go:156]: This should be an error logged since its an impossible condition

Comment (nalepae) [reply beacon-chain/sync/validate_data_column.go]: Fixed in https://github.com/prysmaticlabs/prysm/pull/14559/commits/843b43d309a9205a08dd55c43603a866ad81b133.

**Pull request merged**